File: src/lib/tauri-api/workspace-readiness.ts
import type { WorkspaceReadiness } from '../../types';
import { invokeCommand } from './client';

export function getWorkspaceReadiness(workspaceId: string): Promise<WorkspaceReadiness> {
  return invokeCommand<WorkspaceReadiness>('get_workspace_readiness', { workspaceId });
}

File: src-tauri/src/commands/workspace_readiness.rs
use tauri::State;

use crate::models::WorkspaceReadiness;
use crate::services::workspace_readiness_service;
use crate::state::AppState;

#[tauri::command]
pub fn get_workspace_readiness(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<WorkspaceReadiness, String> {
    workspace_readiness_service::get_workspace_readiness(&state, &workspace_id)
}
